test/wanda_web/controllers/v1/catalog_controller_test.exs

```diff
@@ -34,5 +34,15 @@ defmodule WandaWeb.V1.CatalogControllerTest do
       api_spec = ApiSpec.spec()
       assert_schema(json, "CatalogResponse", api_spec)
     end
+
+    test "does not accept different types inside the env", %{conn: conn} do
+      json =
+        conn
+        |> get("/api/v1/checks/catalog?provider=azure&foo=true")
+        |> json_response(422)
+
+      api_spec = ApiSpec.spec()
+      assert_schema(json, "CatalogResponse", api_spec)
```

Comment:
In case of an error, I believe we should get something different than a `CatalogResponse`.
We don't have, yet, an `UnprocessableEntity` like we do in web, but we could at least assert that the json response represents an error
```elixir
assert %{
          "errors" => [
            %{
              "detail" => ...,
              "title" => ...,
              "source" => ...
            }
          ]
        } = json
```